SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LBDataStorage.cpp

```diff
@@ -19,6 +19,28 @@ void FLBDataCollection::Advance()
 		{
 		case EntityDelta::UPDATE:
 		{
+			for (const auto& Added : Delta.ComponentsAdded)
+			{
+				for (auto& Storage : DataStorages)
+				{
+					if (Storage->GetComponentsToWatch().Contains(Added.ComponentId))
+					{
+						Storage->OnComponentAdded(Delta.EntityId, Added.ComponentId, Added.Data);
+					}
+				}
+			}
+
+			for (const auto& Added : Delta.ComponentsRemoved)
+			{
+				for (auto& Storage : DataStorages)
+				{
+					if (Storage->GetComponentsToWatch().Contains(Added.ComponentId))
+					{
+						// The removal is not communicated though...
```

Comment:
This is a utility to process updates and spit out a simpler format for the data.
Most if this data is tied to SpatialView and Sub View completeness.
Addition and updates end up reflected in the "Modified" Set of the corresponding storage. Removing an individual component clear the data entry and the modified set.
Entity addition and deletion end up in the Added/Removed array of the encompassing class.
So what I mean is that there is no equivalent storage to record the event "this entity has lost this component". Which we might not need, but since it's the only "lost" event, I figured I would at least call it out.
So far there was no dynamic component addition, most data is tied to the existence of the entity. And maybe this is how it should be.

Reply:
Thanks for the clarification - can we improve the code comments with something online these lines but briefer?

Reply:
+1